vendor/bat-native-ads/src/bat/ads/internal/covariates/log_entries/notification_ad_event.h

```diff
@@ -3,33 +3,39 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-#ifndef BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_COVARIATES_LOG_ENTRIES_NOTIFICATION_AD_CLICKED_H_
-#define BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_COVARIATES_LOG_ENTRIES_NOTIFICATION_AD_CLICKED_H_
+#ifndef BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_COVARIATES_LOG_ENTRIES_NOTIFICATION_AD_EVENT_H_
+#define BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_COVARIATES_LOG_ENTRIES_NOTIFICATION_AD_EVENT_H_
 
 #include <string>
 
 #include "bat/ads/internal/covariates/covariate_log_entry_interface.h"
 
 namespace ads {
 
-class NotificationAdClicked final : public CovariateLogEntryInterface {
+class NotificationAdEvent final : public CovariateLogEntryInterface {
  public:
-  NotificationAdClicked();
-  NotificationAdClicked(const NotificationAdClicked&) = delete;
-  NotificationAdClicked& operator=(const NotificationAdClicked&) = delete;
-  ~NotificationAdClicked() override;
+  enum Event {
+    kClicked,
+    kDismissed,
+    kTimedOut
```

Comment:
maybe use `kTimeout` analogous to the confirmation type (despite it being a noun)